Sup 37 get warehouse info

alexeyoverchenko opened the pull request:
for review

alexeyoverchenko committed:
create "get warehouse info" functional

alexeyoverchenko committed:
create "get warehouse info" functional

alexeyoverchenko committed:
get Warehouse info

svareyko commented on src/main/java/by/itech/lab/supplier/controller/WarehouseController.java:
```diff
@@ -1,13 +1,37 @@
+    @GetMapping
+    public Page<WarehouseDto> findAllByActive(
+            @PageableDefault(sort = {"id"}, direction = Sort.Direction.DESC) final Pageable pageable,
+            @RequestParam(required = false) final Boolean status) {
```
maybe would be simpler rename to 'active'?

alexeyoverchenko replied:
it's name from BaseServise interface. we use the same names

svareyko replied:
maybe need to rename there too?

svareyko commented on src/main/java/by/itech/lab/supplier/repository/UserRepository.java:
```diff
@@ -16,7 +16,7 @@
 
     @Modifying
-    @Query("update User set activeForDeletion = false where id = :id")
+    @Query("update User set deleted = false where id = :id")
```
from the logic side: if `deleted == true` - its not deleted, but if `false` - its deleted. is it ok?

alexeyoverchenko replied:
Yes, we all should fix it in our classes. we'll talk about it tomorrow

svareyko replied:
as per our discussion everyone who will change something near to places where its should be renamed - should rename it

svareyko commented on src/main/java/by/itech/lab/supplier/repository/WarehouseRepository.java:
```diff
@@ -0,0 +1,10 @@
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+public interface WarehouseRepository extends JpaRepository<Warehouse, Long> {
```
Should it be marked as `@Repository?`

alexeyoverchenko replied:
done

svareyko replied:
I cant see you "done" until you have not uploaded it here

svareyko commented on src/main/java/by/itech/lab/supplier/service/impl/WarehouseServiceImpl.java:
```diff
@@ -0,0 +1,45 @@
+
+    @Override
+    public WarehouseDto save(WarehouseDto dto) {
+        return null;
```
?

alexeyoverchenko replied:
I've implemented BaseServer methods. "save" will need it to the future

svareyko replied:
so, mark as TODO

alexeyoverchenko committed:
fix all moments except renaming "findById" on "active", cuz it's the general name of the method

svareyko reviewed (approved):
Please rename status to active in the controller and then it can be merged

alexeyoverchenko committed:
- add create/edit/delete methods

alexeyoverchenko committed:
- removed todo
- made changeset (removed "not" null for address_id in User table)
- made changeset (removed "not" null for birthday in User table)

IrynaZazyba commented on src/main/java/by/itech/lab/supplier/service/impl/WarehouseServiceImpl.java:
```diff
@@ -0,0 +1,48 @@
+
+    @Override
+    public Page<WarehouseDto> findAllByActive(Pageable pageable, Boolean active) {
+        return warehouseRepository.findAll(pageable).map(warehouseMapper::map);
```
According to TR  Warehouses can't be enable or disable

alexeyoverchenko replied:
I've corrected it

alexeyoverchenko committed:
- finished Warehouse get info methods

alexeyoverchenko committed:
- finished Warehouse get info methods

IrynaZazyba commented on src/main/java/by/itech/lab/supplier/dto/mapper/WarehouseMapper.java:
```diff
@@ -0,0 +1,40 @@
+
+@AllArgsConstructor
+public class WarehouseMapper implements BaseMapper<Warehouse, WarehouseDto> {
+
```
Maybe you lost @Component

IrynaZazyba commented on src/main/java/by/itech/lab/supplier/domain/Warehouse.java:
```diff
@@ -14,83 +25,27 @@
-@Table(name = "warehouse")
-public class Warehouse {
+@Table
+@Where(clause = "deleted_at is null")
```
You don't realize such functionality, remove it

alexeyoverchenko committed:
- resolved HashSet problem in Warehouse
- deleted @Where annotation
- deleted OneToOne annotation in Category

alexeyoverchenko committed:
Merge branch 'develop' into sup-37_Get_wrehouse_info

# Conflicts:
#	src/main/java/by/itech/lab/supplier/domain/Warehouse.java
#	src/main/java/by/itech/lab/supplier/repository/UserRepository.java

alexeyoverchenko committed:
- resolved HashSet problem in Warehouse
- deleted @Where annotation
- deleted OneToOne annotation in Category

alexeyoverchenko committed:
- return BaseRepository method to default name

IrynaZazyba commented on src/main/java/by/itech/lab/supplier/service/impl/CustomerServiceImpl.java:
```diff
@@ -24,7 +24,7 @@
     @Override
-    public Page<CustomerDto> findAllByActive(final Pageable pageable, final Boolean status) {
+    public Page<CustomerDto> findAll(final Pageable pageable, final Boolean status) {
         return customerRepository.findByStatus(pageable, status).map(customerMapper::mapToCustomerView);
```
I don't understand why you rename other services

IrynaZazyba commented on src/main/java/by/itech/lab/supplier/service/impl/WarehouseServiceImpl.java:
```diff
@@ -0,0 +1,47 @@
+    public WarehouseDto findById(Long warehouseId) {
+        return warehouseRepository.findById(warehouseId).map(warehouseMapper::map)
+                .orElseThrow(NotFoundInDBException::new);
+    }
```
If I try to get a resource which doesn't exist I'm getting 500 error. You throw an exception but never handling it

IrynaZazyba commented on src/main/java/by/itech/lab/supplier/service/impl/WarehouseServiceImpl.java:
```diff
@@ -0,0 +1,47 @@
+    @Override
+    public Page<WarehouseDto> findAllByActive(Pageable pageable, Boolean active) {
+        return null;
+    }
```
null?

alexeyoverchenko committed:
Merge branch 'develop' into sup-37_Get_wrehouse_info

alexeyoverchenko committed:
- in progress

alexeyoverchenko committed:
- resolved naming problem, connected with findAll method, when expected findAllByActive
- findAllByActive() sended in local Services and removed from BaseService
- replaced NotFoundInDBException throw by ResourceNotFoundException in findById

alexeyoverchenko committed:
- resolved naming problem, connected with findAll method, when expected findAllByActive
- findAllByActive() sended in local Services and removed from BaseService
- replaced NotFoundInDBException throw by ResourceNotFoundException in findById

alexeyoverchenko committed:
- resolved naming problem, connected with findAll method, when expected findAllByActive
- findAllByActive() sended in local Services and removed from BaseService
- replaced NotFoundInDBException throw by ResourceNotFoundException in findById
- also add @Repository annotation to all local repositoryes

alexeyoverchenko committed:
Merge branch 'develop' into sup-37_Get_wrehouse_info

alexeyoverchenko committed:
- resolved naming problem, connected with findAll method, when expected findAllByActive
- findAllByActive() sended in local Services and removed from BaseService
- replaced NotFoundInDBException throw by ResourceNotFoundException in findById

alexeyoverchenko committed:
- resolved naming problem, connected with findAll method, when expected findAllByActive
- findAllByActive() sended in local Services and removed from BaseService
- replaced NotFoundInDBException throw by ResourceNotFoundException in findById

alexeyoverchenko committed:
- done for merge
